### PR TITLE
Compact compiler error on MacOs

### PR DESCRIPTION
**Pull request by akeles85**
After compact merge, client was not able to compile on Macos. This PR contains changes to compile it on MacOs.

### CODE REVIEW

**Review by OzanCansel — DISMISSED** [review]

File: hazelcast/test/src/HazelcastTests2.cpp [hunk 1]
```diff
@@ -786,7 +786,42 @@ TEST_F(PipeliningTest, testPipeliningFunctionalityDepth100)
 namespace hazelcast {
 namespace client {
 namespace test {
-class PortableVersionTest : public ::testing::Test
+
+class serializatation_test_base : public testing::Test
```
Comment (OzanCansel) [inline hazelcast/test/src/HazelcastTests2.cpp:790]:
```suggestion
class serialization_test_base : public testing::Test
```

File: hazelcast/test/src/HazelcastTests2.cpp [hunk 2]
```diff
@@ -1010,8 +1046,15 @@ TEST_F(JsonValueSerializationTest, testSerializeDeserializeJsonValue)
     ASSERT_EQ(jsonValue, jsonDeserialized.value());
 }
 
-class ClientSerializationTest : public ::testing::Test
+class ClientSerializationTest : public ClientTest
```
Comment (OzanCansel) [inline hazelcast/test/src/HazelcastTests2.cpp:1049]:
```suggestion
class ClientSerializationTest : public serialization_test_base
```

**Review by OzanCansel — CHANGES_REQUESTED** [review]

File: hazelcast/test/src/HazelcastTests2.cpp (hunk 2 above)
Comment (akeles85) [reply hazelcast/test/src/HazelcastTests2.cpp]: fixed

File: hazelcast/test/src/HazelcastTests2.cpp (hunk 1 above)
Comment (akeles85) [reply hazelcast/test/src/HazelcastTests2.cpp]: fixed

**Review by OzanCansel — APPROVED** [review]

**Pull request merged**